src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/ApplicationDesignerWindowPane.vb

```diff
@@ -71,6 +71,9 @@ Namespace Microsoft.VisualStudio.Editors.ApplicationDesigner
                 'Must marshal ourselves as IUnknown wrapper
                 VSErrorHandler.ThrowOnFailure(WindowFrame.SetProperty(__VSFPROPID.VSFPROPID_ViewHelper, New UnknownWrapper(_viewHelper)))
 
+                'Don't reopen the designer during solution restore
+                VSErrorHandler.ThrowOnFailure(WindowFrame.SetProperty(__VSFPROPID5.VSFPROPID_DontAutoOpen, True))
```

Comment:
Clearly legacy is doing this a different way; I'd be worried about branch switches that reloaded the solution - does this mean that the designer will now close in this situation?

This fix (IMO) is worse than the "bug".

Reply:
I just tested legacy; the AppDesigner is restored if it active, not if it is a background tab.

Reply:
> I'd be worried about branch switches that reloaded the solution

Indeed the design does close when doing branch switches, which is not ideal. Legacy has the same behaviour if there are changes to the project file across branches (and possibly other scenarios)